File: support_tools/swigwin-3.0.12/Examples/test-suite/ruby/director_exception_runme.rb
#!/usr/bin/env ruby
#
# Put description here
#
# 
# 
# 
#

require 'swig_assert'

require 'director_exception'

include Director_exception

class MyFoo < Foo
  def ping
    raise NotImplementedError, "MyFoo::ping() EXCEPTION"
  end
end

class MyFoo2 < Foo
  def ping
    nil # error: should return a string
  end
end

class MyFoo3 < Foo
  def ping
    5 # error: should return a string
  end
end

ok = false

a = MyFoo.new
b = launder(a)

begin
  b.pong
rescue NotImplementedError
  ok = true
end

raise RuntimeError unless ok

ok = false

a = MyFoo2.new
b = launder(a)

begin
  b.pong
rescue TypeError
  ok = true
end


a = MyFoo3.new
b = launder(a)

begin
  b.pong
rescue TypeError
  ok = true
end


raise RuntimeError unless ok

File: support_tools/swigwin-3.0.12/Examples/test-suite/perl5/director_exception_runme.pl
use strict;
use warnings;
use Test::More tests => 7;
BEGIN { use_ok 'director_exception' }
require_ok 'director_exception';

{
  package MyFoo;
  use base 'director_exception::Foo';
  sub ping {
    die "MyFoo::ping() EXCEPTION";
  }
}
{
  package MyFoo2;
  use base 'director_exception::Foo';
  sub ping {
    # error should return a string
    return bless [ 1 ], 'main';
  }
}
{
  package MyFoo3;
  use base 'director_exception::Foo';
  sub ping {
    # error should return a string
    return sub { 1 }
  }
}

{
  my $a = MyFoo->new();
  my $b = director_exception::launder($a);
  eval { $b->pong() };
  like($@, qr/\bMyFoo::ping\(\) EXCEPTION\b/,
      'MyFoo.pong() error content preserved');
}
{
  my $a = MyFoo2->new();
  my $b = director_exception::launder($a);
  eval { $b->pong() };
  like($@, qr/\bTypeError\b/,
      'MyFoo2.pong() error content preserved');
}
{
  my $a = MyFoo3->new();
  my $b = director_exception::launder($a);
  eval { $b->pong() };
  like($@, qr/\bTypeError\b/,
      'MyFoo2.pong() error content preserved');
}

eval { die director_exception::Exception1->new() };
isa_ok($@, 'director_exception::Exception1', 'Exception1');

eval { die director_exception::Exception2->new() };
isa_ok($@, 'director_exception::Exception2', 'Exception2');
